stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -120,6 +133,38 @@ boolean isEmpty() {
     }
   }
 
+  /**
+   * Return the object from the hash map deserialized to the right type. Throw an exception if the
+   * Object cannot be deserialized to that type. This overload is preferred when deserializing
+   * parameterized types as it provides stronger type checking at deserialization time.
+   *
+   * @param <T> - type of class to expect in the hash map
+   * @param key - key to lookup in the hash map
+   * @param typeReference - class requested
+   * @return null if not found or if a null value is stored at that key (use method {@code
+   *     containsKey()} to differentiate)
+   * @throws JsonConversionException if found, not deserializable to the requested type
```

Comment:
might be clearer to say
```suggestion
   * @throws JsonConversionException if found but not deserializable to the requested type
```
Or even
```suggestion
   * @throws JsonConversionException if not deserializable to the requested type
```
Since the missing case is already documented.

Reply:
Fixed on both overloads.